### PR TITLE
Remove SAS Generation Methods from Clients

### LINKED ISSUE #5541: Synchronize Java and .NET Generate SAS Methods
Java and .NET look drastically different with respects to generating SAS tokens. Java has many methods attached to clients to aid in this and this should be changed to follow what .NET has.

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes #5541

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobSasConstructor.java [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+import com.azure.storage.common.credentials.SharedKeyCredential;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.OffsetDateTime;
+
+/**
+ * This class provides a fluent API to help aid in the creation of Azure Blob SAS Tokens.
+ */
+public class BlobSasConstructor {
```
Comment (JonathanGiles) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobSasConstructor.java]: 'Constructor' feels like an unfortunate suffix.

Comment (JonathanGiles) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobSasConstructor.java]: I almost feel compelled to let you call this a Builder with the required CheckStyle suppression.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobSasConstructor.java [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+            sb.append(Utility.urlEncode(param)).append('=').append(Utility.urlEncode(value.toString()));
+        }
+    }
+
+    private String formatQueryDate(OffsetDateTime dateTime) {
+        if (dateTime == null) {
+            return null;
+        } else {
+            return Utility.ISO_8601_UTC_DATE_FORMATTER.format(dateTime);
+        }
```
Comment (JonathanGiles) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobSasConstructor.java]: We ain't paying for all these lines of code! Use ternary operator.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobSasConstructor.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobSasConstructor.java]: I think the `Builder` suffix feels more fluid. Will get additional input on this class before continuing forward on implementing this and using it in Files and Queues.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobSasConstructor.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobSasConstructor.java]: Will cleanup if the class is kept

**Pull request merged**